pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -323,6 +323,12 @@ func (og *operationGenerator) GenerateAttachVolumeFunc(
 				}
 
 			}
+			addVolumeNodeErr := actualStateOfWorld.MarkVolumeAsUncertain(
+				v1.UniqueVolumeName(""), volumeToAttach.VolumeSpec, volumeToAttach.NodeName)
+			if addVolumeNodeErr != nil {
+				// On failure, return error. Caller will log and retry.
+				return volumeToAttach.GenerateError("AttachVolume.MarkVolumeAsUncertain failed", addVolumeNodeErr)
+			}
 			// On failure, return error. Caller will log and retry.
 			return volumeToAttach.GenerateError("AttachVolume.Attach failed", attachErr)
```

Comment:
What about Detach failures? Should they also result in `MarkVolumeAsUncertain`?

Reply:
I don't think so. If detach failed or timed out, attach/detach controller just does not remove the volume from actual state. We don't need to use MarkVolumeAsUncertain to add it since the volume is already in actual state. Then in the next iteration, attach/detach controller just trigger a detach operation again.